Packages/MovieDBUI/Sources/MovieDBUI/UI/Scenes/MovieDetails/MovieDetailsScene.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  MovieDetailsScene.swift
+//  
+//
+//  Created by Mouad Bj on 12/7/2023.
+//
+
+import UIKit
+import MovieDBCore
+
+// a Scene creates and glues the View and the ViewModel
+// it doesn't create depdendencies, as they are passed through constructor (by the composer)
+public class MovieDetailsScene {
+  
+  private let movieDetailsViewController: MovieDetailsViewController
+  
+  public init(repository: MoviesRepository,
```

Comment:
1. Yes ofc! As we discussed, we can use any DI framework with property wrapper injection (@Inject) which will simplify the manual passing of dependencies. 
2. As I said before the only downside to it IMO, is that our whole project will depend strongly on a third-party framework which will be hard to replace if needed (we can discuss solutions for this).  + Most frameworks follow the `Service Locator pattern` which is considered by many an _anti-pattern_, but we can discuss this further separately.
3.  Another thing is, as pretty as the `Scene` object is, it did add multiple steps, if we removed it we will have fewer steps, as the composer will create the viewModel directly and inject all dependencies in one place, but all this could be replaced with a true `DI framework` as u said, this sample project doesn't have any third-party libraries on purpose. However, if we agreed to choose one to be the proper solution then we can implement it in the end.

_// TODO: we need to discuss whether we are going to use a DI framework or an inhouse one (whether through manual constructor injection or Service Locator based)_